dynawo/3rdParty/toolchain.sh

```diff
@@ -103,7 +103,11 @@ compile_nicslu() {
 
 compile_sundials() {
   cd $SCRIPT_DIR/sundials
-  bash sundials-chain-2-7-0.sh --build-dir=$SUNDIALS_BUILD_DIR --install-dir=$SUNDIALS_INSTALL_DIR --suitesparse-install-dir=$SUITESPARSE_INSTALL_DIR --build-type=$BUILD_TYPE
+  if [ "$DYNAWO_USE_SUNDIALS_2_7_0" == "true" ]; then
+    bash sundials-chain-2-7-0.sh --build-dir=$SUNDIALS_BUILD_DIR --install-dir=$SUNDIALS_INSTALL_DIR --suitesparse-install-dir=$SUITESPARSE_INSTALL_DIR --build-type=$BUILD_TYPE
+  else
+    bash sundials-chain-4-0-2.sh --build-dir=$SUNDIALS_BUILD_DIR --install-dir=$SUNDIALS_INSTALL_DIR --suitesparse-install-dir=$SUITESPARSE_INSTALL_DIR --build-type=$BUILD_TYPE
+  fi
```

Comment:
I found it a good idea to ensure the possibility to come back to sundials 2.7.0 if we detect a problem with sundials 4.0.2 to see if it worked beforehands and doesn't work now. Nevertheless, having read a more important part of the PR, it seems that it is not possible to switch easily because the data structure have changed and so you adapted the code into dynawo (especially into the solvers part), which means I can't really go back to Sundials V2.7.0.

The only solution would be to add some conditional compilation but from my point of view, it will lead to too much conditional compilations.

Therefore I would recommend deleting the possibility to use Sundials 2.7.0 here

Reply:
Solved